encfs/BlockFileIO.cpp

```diff
@@ -164,8 +164,8 @@ ssize_t BlockFileIO::read(const IORequest &req) const {
       break;  // didn't get enough bytes
     }
 
-    size_t cpySize = min((size_t)(readSize - partialOffset), size);
-    CHECK(cpySize <= readSize);
+    size_t cpySize = min((size_t)readSize - (size_t)partialOffset, size);
```

Comment:
Shouldn't partialOffset be unsigned?

Reply:
Literally yes, but as req.offset is signed, I would say no.
In addition, AFAIR, min() expect 2 members of the same type, so cast would be required.